backend/handlers/user-handler.go

```diff
@@ -46,17 +46,22 @@ func SignUpHandler(w http.ResponseWriter, r *http.Request) {
     defer cancel()
 
     // Check if the user already exists
-    filter_auth := bson.M{"email": authUser.Email}
+    filter_email := bson.M{"email": authUser.Email}
+    filter_username := bson.M{"username": authUser.Username}
     filter_user := bson.M{"basics.email": authUser.Email}
-    auth_err := authCollection.FindOne(ctx, filter_auth).Decode(&authUser)
+
+    auth_err := authCollection.FindOne(ctx, filter_email).Decode(&authUser)
+    username_err := authCollection.FindOne(ctx, filter_username).Decode(&authUser)
     user_err := userCollection.FindOne(ctx, filter_user).Decode(&user)
     if auth_err == nil {
```

Comment:
Better variable names please...!
And I guess it should be "email already exists"

Reply:
yess sir

Reply:
check please @krishnaacharyaa